brian2/codegen/cpp_prefs.py

```diff
@@ -7,31 +7,40 @@
 
 '''
 from distutils.ccompiler import get_default_compiler
+import platform
 
 from brian2.core.preferences import prefs, BrianPreference
+from brian2.utils.logger import get_logger
+
 from .codeobject import sys_info
 
 __all__ = ['get_compiler_and_args']
 
 # Try to get architecture information to get the best compiler setting for
 # Windows
 msvc_arch_flag = ''
-try:
-    from cpuinfo import cpuinfo
-    res = cpuinfo.get_cpu_info()
-    # Note that this overwrites the arch_flag, i.e. only the best option will
-    # be used
-    if 'sse' in res['flags']:
-        msvc_arch_flag = '/arch:SSE'
-    if 'sse2' in res['flags']:
-        msvc_arch_flag = '/arch:SSE2'
-    if 'avx' in res['flags']:
-        msvc_arch_flag = '/arch:AVX'
-    if 'avx2' in res['flags']:
-        msvc_arch_flag = '/arch:AVX2'
-except Exception:
-    pass  # apparently it does not always work on appveyor
-
+# Only do this on Windows, importing cpuinfo can lead to problems when using
+# multiprocessing (e.g. in PyCharm's debug mode)
+if platform.system() == 'Windows':
+    try:
+        from cpuinfo import cpuinfo
+        res = cpuinfo.get_cpu_info()
+        # Note that this overwrites the arch_flag, i.e. only the best option will
+        # be used
+        if 'sse' in res['flags']:
+            msvc_arch_flag = '/arch:SSE'
+        if 'sse2' in res['flags']:
+            msvc_arch_flag = '/arch:SSE2'
+        if 'avx' in res['flags']:
+            msvc_arch_flag = '/arch:AVX'
+        if 'avx2' in res['flags']:
+            msvc_arch_flag = '/arch:AVX2'
```

Comment:
I wonder if we could entirely eliminate our dependency on py-cpuinfo by just always having ``/arch:SSE /arch:SSE2 /arch:AVX /arch:AVX2`` in the flags for windows? It'll presumably ignore the ones it doesn't know but might use the last of the flags that are applicable? I haven't tried that though.

Reply:
I'd be more than happy to do that, but we'd indeed need to test it first. Maybe you can do it on your machine, which I guess has AVX2, and I do it in the VM, which is probably on the lower end of capabilities?